src/libponyc/type/cap.c

```diff
@@ -402,15 +402,30 @@ bool is_cap_compat_cap(token_id left_cap, token_id left_eph,
   if((left_cap == TK_TAG) || (right_cap == TK_TAG))
     return true;
 
-  // Every possible instantiation of the left cap must be compatible with every
-  // possible instantiation of right cap.
+  // We assume that generic cap constraints came from the same source,
+  // as with is_cap_sub_cap_constraint.
   switch(left_cap)
   {
+    // special-case to keep current behavior
+    // but this is a hack and is part of generic unsoundness
```

Comment:
Can you add a link to the issue ticket in this comment, so that it can be tracked in the future?

Reply:
@jemc is there a comment already? i'm trying to finish off this PR and I'm not sure what ticket this is referring to.

Reply:
@SeanTAllen the issue number is https://github.com/ponylang/ponyc/issues/1931